wolfcrypt/src/asn.c

```diff
@@ -33283,6 +33329,64 @@ static int ParseCRL_Extensions(DecodedCRL* dcrl, const byte* buf,
             }
         #endif
         }
+        else if (oid == CRL_NUMBER_OID) {
+            localIdx = idx;
+            if (GetASNTag(buf, &localIdx, &tag, sz) == 0 &&
+                    tag == ASN_INTEGER) {
+                ret = GetASNInt(buf, &idx, &length, sz);
+                if (ret < 0) {
+                    WOLFSSL_MSG("\tcouldn't parse CRL number extension");
+                    return ret;
+                }
+                else {
+                    if (length > 1) {
+                    #ifdef WOLFSSL_SMALL_STACK
+                        mp_int* m;
+                    #else
+                        mp_int m[1];
+                    #endif
+                        int    i;
+
+                    #ifdef WOLFSSL_SMALL_STACK
+                        m = (mp_int*)XMALLOC(sizeof(*m), NULL,
+                                DYNAMIC_TYPE_BIGINT);
+                        if (m == NULL) {
+                            return MEMORY_E;
+                        }
+                    #endif
+                        if (mp_init(m) != MP_OKAY) {
+                            return MP_INIT_E;
+                        }
+
+                        ret = mp_read_unsigned_bin(m, buf + idx, length);
+                        if (ret != MP_OKAY) {
+                            mp_free(m);
+                    #ifdef WOLFSSL_SMALL_STACK
+                            XFREE(m, NULL, DYNAMIC_TYPE_BIGINT);
+                    #endif
+                            return BUFFER_E;
+                        }
+
+                        dcrl->crlNumber = 0;
+                        for (i = 0; i < (*m).used; ++i) {
+                            if (i > (int)sizeof(word32)) {
+                                    break;
+                            }
+                            dcrl->crlNumber |= ((word32)(*m).dp[i]) <<
+                                (DIGIT_BIT * i);
+                        }
+
+                    #ifdef WOLFSSL_SMALL_STACK
+                        XFREE(m, NULL, DYNAMIC_TYPE_TMP_BUFFER);
```

Comment:
Should be `DYNAMIC_TYPE_BIGINT`

Reply:
Damnit, changed the free above and moved on. Fixed.